extensions/amp-analytics/0.1/events.js

```diff
@@ -151,6 +151,20 @@ function isAmpStoryTriggerType(triggerType) {
   return triggerType.startsWith('story');
 }
 
+function assertUniqueSelectors(selectors) {
+  if (isArray(selectors)) {
+    const map = {};
+    for (let i = 0; i < selectors.length; i++) {
+      userAssert(
+        !map[selectors[i]],
+        'Cannot have duplicate selectors in selectors list: %s',
+        selectors
+      );
+      map[selectors[i]] = selectors[i];
+    }
+  }
```

Comment:
I know you extracted this function from an existing method, but this might be an opportunity to write this more simply:

```suggestion
  userAssert(
    !isArray(selectors) || new Set(selectors).size === selectors.length,
    "Cannot have duplicate selectors in selectors list: %s",
    selectors
  );
```

Reply:
When did we allow Sets in our codebase?

Reply:
Was it ever forbidden? There's a few usages already.

In any case, I missed that `new Set(iterable)` [is not supported on Internet Explorer.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/Set#browser_compatibility)

We can still use it, but [I suggested](https://github.com/ampproject/amphtml/pull/34841#discussion_r658020371) adding a clause to degrade gracefully.